src/core/ipe-setup.c

```diff
@@ -0,0 +1,110 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "copy.h"
+#include "dirent-util.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "ipe-setup.h"
+#include "path-util.h"
+
+#define IPE_SECFS_DIR "/sys/kernel/security/ipe"
+#define IPE_SECFS_NEW_POLICY IPE_SECFS_DIR "/new_policy"
+#define IPE_SECFS_POLICIES IPE_SECFS_DIR "/policies/"
+#define IPE_INPUT_POLICY_DIR "/etc/ipe"
+
+int ipe_setup(void) {
+#if ENABLE_IPE
+        _cleanup_closedir_ DIR *policy_dir = NULL;
+        int r;
+
+        /* Very quick smoke tests first: this is in the citical, sequential boot path, and in most cases it
+         * is unlikely this will be configured, so do the fastest existence checks first and immediately
+         * return if there's nothing to do. */
+
+        if (access(IPE_SECFS_DIR, F_OK) < 0) {
+                log_debug_errno(errno, "IPE support is disabled in the kernel, ignoring: %m");
+                return 0;
+        }
+
+        if (access(IPE_INPUT_POLICY_DIR, F_OK) < 0) {
+                log_debug_errno(errno,
+                                "No IPE custom policy directory %s, ignoring: %m",
+                                IPE_INPUT_POLICY_DIR);
+                return 0;
+        }
+
+        policy_dir = opendir(IPE_INPUT_POLICY_DIR);
+        if (!policy_dir)
+                return log_error_errno(
+                                errno,
+                                "Failed to open the IPE custom policy directory %s: %m",
+                                IPE_INPUT_POLICY_DIR);
+
+        FOREACH_DIRENT(de, policy_dir, return log_error_errno(errno, "Failed to read %s directory: %m", IPE_INPUT_POLICY_DIR)) {
```

Comment:
Hmm, I think we shouldn't use just /etc/ here. In my PoV policies are not configuration, and hence should be placed in /usr/ really.

Hence, use conf_files_list() or a related call, to get the usual override semantics between /etc/, /run/, /usr/lib/?

Reply:
But we don't load from /usr for IMA or any other policies? And there's no /run/ when this is loaded, so that would be quite pointless I'm sure...

Reply:
> But we don't load from /usr for IMA or any other policies? And there's no /run/ when this is loaded, so that would be quite pointless I'm sure...

We should support it for IMA and the others too. Do not use IMA as stellar example for anything please. 

Why wouldn't there be /run/? SELinux policies are usually activated at initrd→host transition, and people might want to do the same for IPE. That means /run/ might very well be already populated by something, with a dynamic policy.

Reply:
Pretty sure it's never going to be used, but ok